server/issue.go

```diff
@@ -38,6 +40,104 @@ func makePost(userID, channelID, message string) *model.Post {
 	}
 }
 
+func (p *Plugin) httpOAuthConnect(w http.ResponseWriter, r *http.Request, instanceID types.ID) (int, error) {
+	code := r.URL.Query().Get("code")
+	userID := r.URL.Query().Get("state")
+
+	conf := p.getConfig()
+	jsonData, _ := json.Marshal(map[string]string{
+		"grant_type":    "authorization_code",
+		"client_id":     conf.JiraAuthAppClientID,
+		"client_secret": conf.JiraAuthAppClientSecret,
+		"code":          code,
+		"redirect_uri":  fmt.Sprintf("%s/oauth/connect", p.GetPluginURL()),
+	})
+
+	request, err := http.NewRequest(
+		"POST",
+		"https://auth.atlassian.com/oauth/token",
+		bytes.NewBuffer(jsonData),
+	)
+	if err != nil {
+		return respondErr(w, http.StatusBadRequest,
+			errors.Wrap(err, "Error creating get token request onauth2"))
+	}
+
+	request.Header.Set("Content-type", "application/json; charset=UTF-8")
+
+	client := &http.Client{}
+	response, err := client.Do(request)
```

Comment:
I believe this can be done with `oauth2.Config`'s `Exchange` method, rather than making sending a HTTP request manually.

Here's an example from the GitHub plugin: https://github.com/mattermost/mattermost-plugin-github/blob/dd1decf1ae0bd1fad553e4e5fd0a99a573f83c06/server/plugin/api.go#L388. Does the `Exchange` method do what we need here?

Reply:
that's right! I have been playing around with that [here](https://github.com/javaguirre/jira-oauth2-example-golang/blob/master/jira_handlers/oauth_jira.go#L92-L95)